server/sonar-server/src/main/java/org/sonar/server/rule/ws/ActiveRuleCompleter.java

```diff
@@ -76,136 +74,138 @@ public ActiveRuleCompleter(DbClient dbClient, Languages languages) {
 
-  private Collection<String> writeActiveRules(DbSession dbSession, SearchResponse.Builder response, RuleQuery query, List<RuleDto> rules) {
-    Collection<String> qProfileKeys = new HashSet<>();
+  private Set<String> writeActiveRules(DbSession dbSession, SearchResponse.Builder response, RuleQuery query, List<RuleDto> rules) {
+    final Set<String> profileUuids = new HashSet<>();
     Rules.Actives.Builder activesBuilder = response.getActivesBuilder();
 
-    String profileKey = query.getQProfileKey();
-    if (profileKey != null) {
+    QProfileDto profile = query.getQProfile();
+    if (profile != null) {
       // Load details of active rules on the selected profile
-      List<ActiveRuleDto> activeRuleDtos = dbClient.activeRuleDao().selectByProfileKey(dbSession, profileKey);
-      Map<RuleKey, ActiveRuleDto> activeRuleByRuleKey = activeRuleDtos.stream().collect(MoreCollectors.uniqueIndex(d -> d.getKey().ruleKey()));
-      ListMultimap<ActiveRuleKey, ActiveRuleParamDto> activeRuleParamsByActiveRuleKey = activeRuleDtosToActiveRuleParamDtos(dbSession, activeRuleDtos);
+      List<OrgActiveRuleDto> activeRules = dbClient.activeRuleDao().selectByProfile(dbSession, profile);
+      Map<RuleKey, OrgActiveRuleDto> activeRuleByRuleKey = activeRules.stream()
+        .collect(uniqueIndex(ActiveRuleDto::getRuleKey));
+      ListMultimap<ActiveRuleKey, ActiveRuleParamDto> activeRuleParamsByActiveRuleKey = loadParams(dbSession, activeRules);
 
       for (RuleDto rule : rules) {
-        ActiveRuleDto activeRule = activeRuleByRuleKey.get(rule.getKey());
+        OrgActiveRuleDto activeRule = activeRuleByRuleKey.get(rule.getKey());
         if (activeRule != null) {
-          qProfileKeys = writeActiveRules(rule.getKey(), singletonList(activeRule), activeRuleParamsByActiveRuleKey, activesBuilder);
+          profileUuids.addAll(writeActiveRules(rule.getKey(), singletonList(activeRule), activeRuleParamsByActiveRuleKey, activesBuilder));
         }
       }
     } else {
       // Load details of all active rules
-      List<ActiveRuleDto> activeRuleDtos = dbClient.activeRuleDao().selectByRuleIds(dbSession, query.getOrganizationUuid(), Lists.transform(rules, RuleDto::getId));
-      Multimap<RuleKey, ActiveRuleDto> activeRulesByRuleKey = from(activeRuleDtos).index(d -> d.getKey().ruleKey());
-      ListMultimap<ActiveRuleKey, ActiveRuleParamDto> activeRuleParamsByActiveRuleKey = activeRuleDtosToActiveRuleParamDtos(dbSession, activeRuleDtos);
+      List<Integer> ruleIds = Lists.transform(rules, RuleDto::getId);
+      List<OrgActiveRuleDto> activeRules = dbClient.activeRuleDao().selectByRuleIds(dbSession, query.getOrganization(), ruleIds);
+      Multimap<RuleKey, OrgActiveRuleDto> activeRulesByRuleKey = activeRules.stream()
+        .collect(MoreCollectors.index(OrgActiveRuleDto::getRuleKey));
+      ListMultimap<ActiveRuleKey, ActiveRuleParamDto> activeRuleParamsByActiveRuleKey = loadParams(dbSession, activeRules);
+      rules.forEach(rule -> {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS00108)